File: app/src/main/java/awais/instagrabber/repositories/responses/directmessages/DirectItemEmojiReaction.java
package awais.instagrabber.repositories.responses.directmessages;

public class DirectItemEmojiReaction {
    private final long senderId;
    private final long timestamp;
    private final String emoji;
    private final String superReactType;

    public DirectItemEmojiReaction(final long senderId, final long timestamp, final String emoji, final String superReactType) {
        this.senderId = senderId;
        this.timestamp = timestamp;
        this.emoji = emoji;
        this.superReactType = superReactType;
    }

    public long getSenderId() {
        return senderId;
    }

    public long getTimestamp() {
        return timestamp;
    }

    public String getEmoji() {
        return emoji;
    }

    public String getSuperReactType() {
        return superReactType;
    }
}

File: app/src/main/java/awais/instagrabber/repositories/responses/directmessages/DirectItemEmojiReaction.kt
package awais.instagrabber.repositories.responses.directmessages

import java.io.Serializable

data class DirectItemEmojiReaction(
    val senderId: Long = 0,
    val timestamp: Long = 0,
    val emoji: String? = null,
    val superReactType: String? = null
) : Serializable